chain/network/src/raw.rs

```diff
@@ -343,3 +267,233 @@ impl Peer {
         Ok(())
     }
 }
+
+mod test {
+    use crate::network_protocol::{
+        Encoding, Handshake, PeerIdOrHash, PeerMessage, Pong, RawRoutedMessage, RoutedMessageBody,
+    };
+    use crate::raw;
+    use crate::time::Utc;
+    use near_crypto::{KeyType, SecretKey};
+    use near_primitives::network::PeerId;
+    use std::io;
+    use tokio::io::{AsyncReadExt, AsyncWrite, AsyncWriteExt};
+    use tokio::net::TcpListener;
+
+    // we add #[allow(dead_code)] to everything here because for some reason this generates
+    // dead code warnings even though everything is very clearly used below... maybe it's this?
+    // https://github.com/rust-lang/rust/issues/46379
```

Comment:
it happens because you haven't added a conditional compilation guard at the top of the module
It should be
```
#[cfg(test)]
mod test { ...}
```
but please move it to a separate file anyway.

Reply:
done